### DISCUSSION TITLE
Mods to test for compatibility

### DISCUSSION THREAD

**Discussion by LevFendi** [Factorio-Access/FactorioAccess#143] [General]
# Will Test Group 1 (Priority)
[AAI Industry](https://mods.factorio.com/mod/aai-industry)
[Fluid Must Flow](https://mods.factorio.com/mod/FluidMustFlow)
[Robot Attrition](https://mods.factorio.com/mod/robot_attrition)
[Shield Projector](https://mods.factorio.com/mod/shield-projector)
[Space Extension Mod (SpaceX)](https://mods.factorio.com/mod/SpaceMod)
[Reverse Factory](https://mods.factorio.com/mod/reverse-factory)
FNEI
Equipment Gantry
Space Exploration
https://mods.factorio.com/mod/armor-spill-prevention 

[Industrial Revolution 3](https://mods.factorio.com/mod/IndustrialRevolution3)
[Dynamic voice TTS](https://mods.factorio.com/mod/Dynamic-voice-TTS)
[Sea Block Pack](https://mods.factorio.com/mod/SeaBlockMetaPack) - Note: Has like 50 required dependencies.

# Will Test Group 2 (Sometime)
[Rampant](https://mods.factorio.com/mod/Rampant)
[Miniloader](https://mods.factorio.com/mod/miniloader)
[Loader Redux](https://mods.factorio.com/mod/LoaderRedux)
[Ultracube: Age of Cube](https://mods.factorio.com/mod/Ultracube)
[Brave New World](https://mods.factorio.com/mod/brave-new-world)

# Not Compatible Group 1 (Will prioritize to make compatible)
[Robotic Voice Sample For Speaker](https://mods.factorio.com/mod/Robotic-Voice-Sample-For-Speaker) - Programmable speaker lists need to be made dynamic

# Note Compatible Group 2 (Just noting the issues for now)
[Simple Seablock](https://mods.factorio.com/mod/SimpleSeablock) - Cannot load the special preset from the FA launcher
[Small assembling](https://mods.factorio.com/mod/Small_assembling) - cannot set recipes as intended
[Krastorio 2](https://mods.factorio.com/mod/Krastorio2) - data stage crash due to inserter collision masks
[Text Plates](https://mods.factorio.com/mod/textplates) - No crashes but the text plates currently cannot be read, crafted, or placed without using GUI, because they all have different prototypes and special crafting and placement rules.

# Partially Compatible Group
(empty)

# Mostly Compatible Group
[Flare Stack](https://mods.factorio.com/mod/Flare%20Stack) - It works but the incineration recipes fill up the crafting menu, even though they cannot be crafted by hand. A fix would be good using an additional check using [this](get_hand_crafting_disabled_for_recipe(recipe))

# Fully Compatible Group
[Simple Air Cleaner](https://mods.factorio.com/mod/simple-air-cleaner)
[Armoured Biters](https://mods.factorio.com/mod/ArmouredBiters)
[Fill4Me](https://mods.factorio.com/mod/Fill4Me)
[Inserter Fuel Leech](https://mods.factorio.com/mod/InserterFuelLeech)

[More on the wiki page, here.](https://github.com/Factorio-Access/FactorioAccess/wiki/Compatible-Other-Mods)

**Comment by ahicks92** [comment]
When I get bored I read mod code.  I will backfill the ones that don't have concerns left in the Wiki later:

- QOL Research: compatible, save for #162.  That's probably a trivial fix for someone who knows where to check. Just uses technology prototypes. Adds some nice to haves like faster hand crafting which don't change the late game much but are useful and make the experience smoother, e.g. hand reach and hand crafting speed.  Don't know how to test that the mod respects the values exposed by the game--I'd not be surprised if we had a hidden constant hiding in the mod where it should be reading and respecting entities.  The associated issue is cosmetic, if annoying and spammy.
- Realistic Reactors: read the code, haven't tried it.  We should support it just about.  It's the standard constant combinator reskin, if we don't we should just need to be able to connect.  We can do other things besides the dragging helpers if it came down to it.  Has a gui, the code does not appear to do anything actionable through it, it's just status and a couple things you could do via the circuit network, and you're *supposed* to do them via the circuit network anyway.  Not something I intend to play with soon but if we support it soething I'd definitely want to play with eventually for the circuit network challenge.
- LTN: if realistic reactors works LTN works and vice versa, again by reading the code and not testing. I might try LTN soon; I'm not yet entirely clear on how I'd use it regardless of accessibility.  They use the same reskinned constant combinator trick.  Some global settings we can't access that might or might not be important, but they can be set on every individual case via the circuit network according to docs.
- Bob's inserters: Also a code read, not a personal test.  No, but kind of yes.   Their GUI is a problem.  Their hotkeys are too limited.  We're probably not set up to present it.  But the actual implementation is just setting the pick-up and drop-off locations of inserters.  Ideally we'd get him to add a remote interface for us because that should be reasonably easy in this case, but we could just duplicate the formulas if it's installed and set it ourselves, that shouldn't be hard.  We can read tech levels from other mods and that's all we really need.

And finally an honorable mention to SpaceX. It has a gui, a lot  of code, but it doesn't appear that the gui is essential based off a skim.  It's a long run though, essentially I think it's "excuse to build a megabase, the mod", we'll probably have to use cheats.

**Comment by ahicks92** [comment]
Dynamic TTS may require integration into the launcher before you can get anywhere with it if it also uses stdout.  Opening a second thing given that we have a first thing required in the same programming language is less than ideal.  We had a whole discussion via discord so that may be known but let's drop it here anyway.

Note that any mod which uses AAI Zones requires using a selection tool but we may be able to work around it because it sort of looked like that adds prototypes we could place.  I think our first interface request should be the ability to invoke selection tools.

**Comment by ahicks92** [comment]
On ltn:

Manual:https://forums.factorio.com/viewtopic.php?t=51072
More informative guide: https://www.reddit.com/r/factorio/comments/73xyd5/guide_for_a_loweffort_ltn_user/

Basically yes it's the circuit network but, to really effectively use it, we probably need all of the following, which are fortunately vanilla features we want to support anyway:

(most of this would also apply to Realistic Reactors)

- Train stops on circuit network.
- Arithmetic combinators (used for doing some binary encoding and decoding; see the guide on smart refueling).
- Full access to train schedules.
- If we don't circuit networks for lamps, just enough to connect.
- Possibly ability to see cargo wagon slots.
- Possibly the decider combinator.  I'd have to play with it to know for  sure.


Without at least some of that I can't test further to any meaningful degree.

Good news, though: they offer a remote interface that can do probably everything the GUI can.
